File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/MallpayInitRequest.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;

@EqualsAndHashCode(callSuper = false)
@Data
@AllArgsConstructor @RequiredArgsConstructor
@JsonInclude(value = JsonInclude.Include.NON_NULL)
public class MallpayInitRequest  extends SignBase {
    @NonNull
    private String merchantId;
    @NonNull
    private String orderNo;
    @NonNull
    private MallpayCustomer customer;
    @NonNull
    private MallpayOrder order;
    @NonNull
    private Boolean agreeTC;
    @NonNull
    private String clientIp;
    @NonNull
    private String returnUrl;
    @NonNull
    private String returnMethod;
    private String merchantData;
    private Long ttlSec;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, merchantId);
        add(sb, orderNo);
        add(sb, customer.toSign());
        add(sb, order.toSign());
        add(sb, agreeTC);
        add(sb, dttm);
        add(sb, clientIp);
        add(sb, returnUrl);
        add(sb, returnMethod);
        add(sb, merchantData);
        add(sb, ttlSec);
        deleteLast(sb);
        return sb.toString();
    }
}

File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/MallpayInitRequest.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class MallpayInitRequest : SignBaseRequest
{
    [JsonProperty("orderNo")] public string? OrderNo { get; set; }

    [JsonProperty("customer")] public MallpayCustomer? Customer { get; set; }

    [JsonProperty("order")] public MallpayOrder? Order { get; set; }

    [JsonProperty("agreeTC")] public bool? AgreeTC { get; set; }

    [JsonProperty("clientIp")] public string? ClientIp { get; set; }

    [JsonProperty("returnUrl")] public string? ReturnUrl { get; set; }

    [JsonProperty("returnMethod")] public string? ReturnMethod { get; set; }

    [JsonProperty("merchantData")] public string? MerchantData { get; set; }

    [JsonProperty("ttlSec")] public long? TtlSec { get; set; }


    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, MerchantId);
        Add(sb, OrderNo);
        if (null != Customer) Add(sb, Customer.ToSign());
        if (null != Order) Add(sb, Order.ToSign());
        Add(sb, AgreeTC);
        Add(sb, Dttm);
        Add(sb, ClientIp);
        Add(sb, ReturnUrl);
        Add(sb, ReturnMethod);
        Add(sb, MerchantData);
        Add(sb, TtlSec);
        DeleteLast(sb);
        return sb.ToString();
    }
}